tests/core/caching-utils/test_request_caching.py

```diff
@@ -211,11 +215,11 @@ def test_all_providers_do_not_cache_by_default_and_can_set_caching_properties(pr
         ("0x5", False),
     ),
 )
-def test_blocknum_validation_against_validation_threshold_when_caching(
+def test_blocknum_validation_against_validation_threshold_when_caching_mainnet(
     threshold, endpoint, blocknum, should_cache, request_mocker
 ):
     w3 = Web3(
-        HTTPProvider(
+        BaseProvider(
```

Comment:
more lazy questions: Is it just the HTTPProvider that caches? or should we maybe have a test that parametrizes the other providers too?

Reply:
Nope, all providers should cache. I think that would be a good idea. Should we track it separately though?

Reply:
We have a [test for caching properties](https://github.com/ethereum/web3.py/blob/46913500aba08113aad07c50b858aca18cc70f46/tests/core/caching-utils/test_request_caching.py#L167-L199) but we don't run all providers through the test suite currently.

Reply:
> Nope, all providers should cache. I think that would be a good idea. Should we track it separately though?

Sure, that's fine by me if you'd rather track separately.

Reply:
I added it in here instead, as a different commit.